assets/scripts/oauth-services-api/index.js

```diff
@@ -0,0 +1,49 @@
+import page from 'page'
+import store from './../store.js'
+import { GitHubAPI } from './github.js'
+
+import './../types.js'
+
+export class OAuthProvider {
+  /**
+   *
+   * @param {string} accessToken
+   * @param {string} oAuthProviderName
+   */
+  constructor(accessToken, oAuthProviderName) {
+    this._accessToken = accessToken
+    this._oAuthProviderName = oAuthProviderName
+    /** @type {OAuthServiceAPI | null} */
+    this._serviceAPI = null
+  }
+
+  /**
+   * @returns {OAuthServiceAPI}
+   */
+  getServiceAPI() {
+    if (!this._accessToken) {
+      page.redirect('/account')
+    }
+
+    if (this._serviceAPI) {
+      return this._serviceAPI
+    }
+
+    // TODO: add other OAuth providers here when needed
+    return new GitHubAPI(this._accessToken)
```

Comment:
Je pense que ça vaudrait le coup de faire un `if(this._oAuthProviderName === 'github')` et de `throw` au `else`

Reply:
commentaire obsolète